src/commands/app/use.js

```diff
@@ -190,37 +190,84 @@ class Use extends BaseCommand {
    * @returns {Buffer} the Adobe Developer Console configuration file for the workspace
    */
   async selectTargetWorkspaceInProject (consoleCLI, config, flags) {
-    const project = { name: config.project.name, id: config.project.id }
-    const currentWorkspace = { name: config.workspace.name, id: config.workspace.id }
+    const workspaceNameFlag = flags.workspace
+    const workspaceIdFlag = flags.workspaceId
+    // workspace flags
+    const workspaceDetails = {
+      workspaceName: workspaceNameFlag,
+      workspaceId: workspaceIdFlag
+    }
+    // workspaceName flag or workspaceId flag
+    const workspaceNameOrId = workspaceNameFlag || workspaceIdFlag
+
+    const { project, org, workspace: currentWorkspace } = config
 
-    // make sure user is not trying to switch to current workspace
-    const workspaceFlag = flags.workspace
-    if (workspaceFlag === currentWorkspace.name) {
+    // 1. Check if the current selected workspace is the same as the requested one
+    if (this.workspaceExist(currentWorkspace, workspaceDetails)) {
       this.cleanConsoleCLIOutput()
-      this.error(`--workspace=${workspaceFlag} is the same as the currently selected workspace, nothing to be done.`)
+      this.error(`--workspace=${workspaceNameOrId} is the same as the currently selected workspace, nothing to be done.`)
     }
 
     // retrieve all workspaces
     const workspaces = await consoleCLI.getWorkspaces(
-      config.org.id,
-      config.project.id
+      org.id,
+      project.id
     )
     const workspacesButCurrent = workspaces.filter(w => w.id !== currentWorkspace.id)
-
     let workspace
 
-    if (workspaceFlag) {
-      // workspace name is given, make sure the workspace is in there
-      workspace = workspacesButCurrent.find(w => w.name === workspaceFlag)
-      if (!workspace) {
-        this.cleanConsoleCLIOutput()
-        this.error(`--workspace=${workspaceFlag} does not exist in current Project ${project.name}.`)
+    if (workspaceNameOrId) {
+      // 2. Find a workspace with the given name or id
+      workspace = workspacesButCurrent.find(currentWorkspace => this.workspaceExist(currentWorkspace, workspaceDetails))
+      if (!workspace && workspaceNameFlag) {
+        aioLogger.debug(`--workspace=${workspaceNameOrId} was not found in the current Project ${project.name}.
+Creating workspace ${workspaceNameOrId}...`)
+        const skipPrompt = !!flags['confirm-new-workspace']
+        // 2.1. Create the workspace with the given name.
+        if (!skipPrompt) {
+          const shouldNewWorkspace = await consoleCLI.prompt.promptConfirm(`Workspace ${workspaceNameFlag} was not found \n > Do you wish to create a new workspace named: '${workspaceNameFlag}'`)
```

Comment:
nitpick: `Workspace '${workspaceNameOrIdFlag}' does not exist \n > Do you wish to create a new workspace ?`